src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,54 +1,123 @@
-import React from 'react';
+import React, { useState } from 'react';
+import cn from 'classnames';
 import './FindMovie.scss';
+import { getMovie } from '../../api/api';
+import { MovieCard } from '../MovieCard';
+import { Movie } from '../../react-app-env';
 
-// import { MovieCard } from '../MovieCard';
-
-export const FindMovie: React.FC = () => (
-  <>
-    <form className="find-movie">
-      <div className="field">
-        <label className="label" htmlFor="movie-title">
-          Movie title
-        </label>
-
-        <div className="control">
-          <input
-            type="text"
-            id="movie-title"
-            placeholder="Enter a title to search"
-            className="input is-danger"
-          />
-        </div>
+type Props = {
+  movies: Movie[],
+  setMovies: React.Dispatch<React.SetStateAction<Movie[]>>,
+};
 
-        <p className="help is-danger">
-          Can&apos;t find a movie with such a title
-        </p>
-      </div>
-
-      <div className="field is-grouped">
-        <div className="control">
-          <button
-            type="button"
-            className="button is-light"
-          >
-            Find a movie
-          </button>
-        </div>
+export const FindMovie: React.FC<Props> = React.memo(
+  ({ movies, setMovies }) => {
+    const [movieTitle, setMovieTitle] = useState('');
+    const [movie, setMovie] = useState<Movie | null>(null);
+    const [hasSearchError, setSearchError] = useState(false);
+    const [hasListError, setListError] = useState(false);
+
+    const loadMovie = async () => {
+      setListError(false);
+      setSearchError(false);
+      setMovieTitle('');
+      const movieFromServer: Movie = await getMovie(movieTitle);
+
+      if (!movieFromServer.Title) {
+        setSearchError(true);
+        setMovie(null);
+      } else {
+        setMovie(movieFromServer);
+      }
+    };
+
+    const pushMovie = () => {
+      if (movie) {
+        setMovie(null);
+        const movieIsUnique = !(movies
+          .some(mov => mov.imdbID === movie.imdbID));
+
+        if (movieIsUnique) {
+          setMovies([...movies, movie]);
+        } else {
+          setListError(true);
+        }
+      }
+    };
+
+    return (
+      <>
+        <form className="find-movie">
```

Comment:
please implement `onSubmit` so that page does not reload and purge the entire list when hitting 'Enter' on keyboard